go.mod

```diff
@@ -3,27 +3,33 @@ module github.com/pressly/goose/v3
 go 1.17
 
 require (
-	github.com/ClickHouse/clickhouse-go/v2 v2.2.0
-	github.com/avast/retry-go/v4 v4.3.0
+	github.com/ClickHouse/clickhouse-go/v2 v2.3.0
```

Comment:
The latest version (`v2.4.3`) isn't compatible with go1.17, because they started using the `net/netip` package, and unfortunately this throws an error:

```
Error: ../../../go/pkg/mod/github.com/!click!house/ch-go@v0.50.0/proto/ipv4.go:5:2: package net/netip is not in GOROOT (/opt/hostedtoolcache/go/1.17.13/x64/src/net/netip)
```

We'll stick with v2.3.0 until go1.20 comes out (~Feb 2023).

Reply:
Oof, upgrading deps isn't fun. Other modules are starting to use generics and transitive deps require go1.18 or higher.

Can't wait until go1.20 lands so the minimum supported version is go1.18. Or maybe we just bite the bullet and make go1.18 the minimum today.